ingestion/src/metadata/interfaces/sqalchemy/sqa_profiler_interface.py

```diff
@@ -369,35 +372,45 @@ def get_all_metrics(
         """get all profiler metrics"""
         logger.info(f"Computing metrics with {self._thread_count} threads.")
         profile_results = {"table": dict(), "columns": defaultdict(dict)}
-        with concurrent.futures.ThreadPoolExecutor(
-            max_workers=self._thread_count
-        ) as executor:
+        with CustomThreadPoolExecutor(max_workers=self._thread_count) as pool:
             futures = [
-                executor.submit(
+                pool.submit(
                     self.compute_metrics_in_thread,
                     *metric_func,
                 )
                 for metric_func in metric_funcs
             ]
 
-        for future in concurrent.futures.as_completed(futures):
-            profile, column, metric_type = future.result()
-            if metric_type != MetricTypes.System.value and not isinstance(
-                profile, dict
-            ):
-                profile = dict()
-            if metric_type == MetricTypes.Table.value:
-                profile_results["table"].update(profile)
-            elif metric_type == MetricTypes.System.value:
-                profile_results["system"] = profile
-            else:
-                profile_results["columns"][column].update(
-                    {
-                        "name": column,
-                        "timestamp": datetime.now(tz=timezone.utc).timestamp(),
-                        **profile,
-                    }
-                )
+            for future in futures:
+                if future.cancelled():
+                    continue
+
+                try:
+                    profile, column, metric_type = future.result(
+                        timeout=self.timeout_seconds
+                    )
+                    if metric_type != MetricTypes.System.value and not isinstance(
+                        profile, dict
+                    ):
+                        profile = dict()
+                    if metric_type == MetricTypes.Table.value:
+                        profile_results["table"].update(profile)
+                    elif metric_type == MetricTypes.System.value:
+                        profile_results["system"] = profile
+                    else:
+                        profile_results["columns"][column].update(
+                            {
+                                "name": column,
+                                "timestamp": datetime.now(tz=timezone.utc).timestamp(),
+                                **profile,
+                            }
+                        )
+                except concurrent.futures.TimeoutError as exc:
+                    pool.shutdown39(wait=True, cancel_futures=True)
+                    logger.debug(traceback.format_exc())
+                    logger.info(f"Operation was cancelled due to TimeoutError - {exc}")
```

Comment:
Not sure if an error is better here:
```suggestion
                    logger.error(f"Operation was cancelled due to TimeoutError - {exc}")
```